test/widgets/autocomplete_test.dart

```diff
@@ -108,24 +108,26 @@ void main() {
   group('ComposeAutocomplete', () {
 
     Finder findNetworkImage(String url) {
+      String trimmedUrl = url.replaceAll(RegExp(r'\.(png|jpg|jpeg|gif|bmp|webp)$',caseSensitive: false), ''); //Remove the extension from the URL
```

Comment:
Ah, one thing I missed last time: we're changing `findNetworkImage` so it's specifically helpful for finding an _avatar_ image, not just any image, so let's change its name accordingly. How about `findAvatarImage`?

I don't think the implementation needs to recite those other extensions ("jpg", etc.). Our code for producing a "-medium" URL only handles `.png`s; see `UploadedAvatarUrl.get` in lib/model/avatar_url.dart.

I also think we can organize the code to be a bit clearer about what it's doing; maybe like this:

```dart
    Finder findAvatarImage(String url) {
      /// Whether an Image widget's URL is for the avatar we're looking for.
      ///
      /// Accepts the -medium.png variant; see [UploadedAvatarUrl.get].
      bool urlMatches(String candidateUrl) =>
        candidateUrl.startsWith(url.replaceAll(RegExp(r'\.png$'), ''));

      return find.byWidgetPredicate((widget) => switch(widget) {
        Image(image: NetworkImage(url: var imageUrl)) when urlMatches(imageUrl)
          => true,
        _ => false,
      });
    }
```

But also, I think there's another way to write `findAvatarImage` without even having to think about the `.png` / `-medium.png` detail. All we want from this `Finder` is to know whether an avatar is shown or not, right? How about this:

```dart
    Finder findAvatarImage(int userId) =>
      find.byWidgetPredicate((widget) => widget is AvatarImage && widget.userId == userId);
```

How about an NFC prep commit that replaces `findNetworkImage` with that?

Reply:
Alright, I'll get started on the changes